packages/vhd-lib/Vhd/VhdFile.js

```diff
@@ -84,7 +84,9 @@ exports.VhdFile = class VhdFile extends VhdAbstract {
 
   static async open(handler, path, { flags, checkSecondFooter = true } = {}) {
     if (handler.isEncrypted) {
-      throw new Error(`VHDFile implementation is not compatible with encrypted remote`)
+      const error = new Error(`VHDFile implementation is not compatible with encrypted remote`)
+      error.code = 'EISENCRYPTED'
+      throw error
```

Comment:
Attempt to open the file before before checking this to properly throw a `EISDIR`.

Close the file in that case and throw.

Add a comment to explain that.